mapie/multi_label_classification.py

```diff
@@ -0,0 +1,738 @@
+            raise ValueError(
+                "Invalid delta. "
+                f"delta must be a float, not a {type(delta)}"
+            )
+        if (delta is not None) and (
+            ((delta <= 0) or (delta >= 1)) and
+            self.method == "rcps"
+        ):
+            raise ValueError(
+                "Invalid delta. "
+                "delta must be in the ]0, 1[ interval"
+            )
+
+    def _check_estimator(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        estimator: Optional[ClassifierMixin] = None,
+    ) -> ClassifierMixin:
+        """
+        Check if estimator is ``None``,
+        and returns a multi-output ``LogisticRegression``
+        instance if necessary.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Training data.
+
+        y : ArrayLike of shape (n_samples,)
+            Training labels.
+
+        estimator : Optional[ClassifierMixin], optional
+            Estimator to check, by default ``None``
+
+        Returns
+        -------
+        ClassifierMixin
+            The estimator itself or a default multi-output
+            ``LogisticRegression`` instance.
+
+        Raises
+        ------
+        ValueError
+            If the estimator is not ``None``
+            and has no fit, predict, nor predict_proba methods.
+
+        NotFittedError
+            If the estimator is not fitted.
+        """
+        if estimator is None:
+            return MultiOutputClassifier(
+                LogisticRegression(multi_class="multinomial")
+            ).fit(X, y)
+
+        if isinstance(estimator, Pipeline):
+            est = estimator[-1]
+        else:
+            est = estimator
+        if (
+            not hasattr(est, "fit")
+            or not hasattr(est, "predict")
+            or not hasattr(est, "predict_proba")
+        ):
+            raise ValueError(
+                "Invalid estimator. "
+                "Please provide a classifier with fit,"
+                "predict, and predict_proba methods."
+            )
+        check_is_fitted(est)
+        return estimator
+
+    def _check_partial_fit_first_call(self) -> bool:
+        """Check that this is the first time partial_fit
+        or fit is called.
+
+        Returns
+        -------
+        bool
+            True if it is the first time, else False.
+        """
+        return not hasattr(self, "risks")
+
+    def _check_bound(self, bound: Optional[str]):
+        """Check the value on the bound.
+
+        Parameters
+        ----------
+        bound : Optional[str]
+            Bound defined in the predict.
+
+        Raises
+        ------
+        AttributeError
+            If bound in not in ["hoeffding", "bernstein", "wsr", None]
+        """
+        if bound not in ["hoeffding", "bernstein", "wsr", None]:
+            raise ValueError(
+                'bound must be in ["hoeffding", "bernstein", "wsr", None]'
```

Comment:
"bound must be in ['hoeffding', 'bernstein', 'wsr', None]"

Reply:
done